src/utils/editor/source-documents.js

```diff
@@ -75,11 +75,9 @@ export function clearEditor(editor: SourceEditor) {
 }
 
 export function showLoading(editor: SourceEditor) {
-  if (hasDocument("loading")) {
-    return;
-  }
+  const docExists = hasDocument("loading");
 
-  const doc = editor.createDocument();
+  const doc = docExists ? getDocument("loading") : editor.createDocument();
```

Comment:
maybe we just want to return if we have the doc...

Reply:
If we just return from there, it doesn't display `loading...` since we are not replacing the editor text, so I see the text of original source and the spinner.

Reply:
got it. maybe get and then replace..